### PR TITLE
fix: unified modal and drawer protal and fix overflow conflict

### LINKED ISSUE #17323: Two scrollbars appears when using modal inside drawer with drawer and body with scroll
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://github.com/SrAnthony/antd-drawer-scroll-bug](https://github.com/SrAnthony/antd-drawer-scroll-bug)

### Steps to reproduce
Click on Open Drawer and then on Open Modal

### What is expected?
When opening the drawer the body scroll should be hidden (ok).
When opening the modal the body and drawer scroll should be hidden (ok).
When closing the modal the body scroll should be kept hidden and the drawer scroll should be visible (not ok)
When closing the drawer the body scroll should be shown (ok).

### What is actually happening?
When closing the modal inside the drawer the body scroll is showing. If both body and drawer have scroll areas, two scrollbars are visible.

| Environment | Info |
|---|---|
| antd | 3.19.8 |
| React | 16.8.1 |
| System | linux |
| Browser | Chrome 75.0.3770.90 |

### PR DESCRIPTION
**Pull request by jljsj33**
### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [X] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |  1. 解决 modal 和 drawer 同时对  body 的 overflow 操作问题, fix #17323 #16680 <br />2. 更新 drawer 和 modal 的 getContainer, 同时支持 HTMLElement \| `() => HTMLElement` \| Selectors \| false, false 为渲染在当前 dom 里  |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

-----
[View rendered components/drawer/index.en-US.md](https://github.com/ant-design/ant-design/blob/merge-portal/components/drawer/index.en-US.md)
[View rendered components/drawer/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/merge-portal/components/drawer/index.zh-CN.md)
[View rendered components/modal/index.en-US.md](https://github.com/ant-design/ant-design/blob/merge-portal/components/modal/index.en-US.md)
[View rendered components/modal/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/merge-portal/components/modal/index.zh-CN.md)

### CODE REVIEW

**Comment by jljsj33** [comment]
为什么 15 会挂?

**Comment by ztplz** [comment]
别搞15了 又没啥人用 都快17了

**Comment by jljsj33** [comment]
我也觉得，，汗。。。

**Comment by jljsj33** [comment]
干他娘的，，怎么又挂了。。。

**Comment by zombieJ** [comment]
3.x 版本里 15 还是要看的。

**Comment by ztplz** [comment]
16那么多好用的都不能用

File: components/drawer/index.zh-CN.md [hunk 1]
```diff
@@ -20,9 +20,9 @@ title: Drawer
 | --- | --- | --- | --- | --- |
 | closable | 是否显示右上角的关闭按钮 | boolean | true | 3.7.0 |
 | destroyOnClose | 关闭时销毁 Drawer 里的子元素 | boolean | false | 3.7.0 |
-| getContainer | 指定 Drawer 挂载的 HTML 节点 | HTMLElement \| `() => HTMLElement` \| Selectors | 'body' | 3.7.0 |
-| maskClosable | 点击蒙层是否允许关闭 | boolean | true | 3.7.0 | 3.7.0 |
-| mask | 是否展示遮罩 | Boolean | true |
+| getContainer | 指定 Drawer 挂载的 HTML 节点, false 为挂载在当前 dom | HTMLElement \| `() => HTMLElement` \| Selectors \| false | 'body' | 3.7.0 |
```
Comment (zombieJ) [inline components/drawer/index.zh-CN.md]: 这应该是新的 feature 了吧？

Comment (jljsj33) [reply components/drawer/index.zh-CN.md]: 不算吧。这 API 本来就存在的，drawer 本来就支持的，，现在是全部都放出来了，，modal 的是跟 drawer 统一了。。不算新功能吧。。

**Comment by jljsj33** [comment]
@afc163 review

**Pull request merged**